src/loaders/VolumeLoaderUtils.ts

```diff
@@ -130,8 +141,33 @@ export function scaleMultipleDimsToSubregion(subregion: Box3, dims: ZYX[]): ZYX[
  */
 export function pickLevelToLoadUnscaled(loadSpec: LoadSpec, spatialDimsZYX: ZYX[]): number {
   const optimalLevel = estimateLevelForAtlas(spatialDimsZYX, loadSpec.maxAtlasEdge);
-  const levelToLoad = Math.max(optimalLevel + (loadSpec.scaleLevelBias ?? 0), loadSpec.multiscaleLevel ?? 0);
-  return Math.max(0, Math.min(spatialDimsZYX.length - 1, levelToLoad));
+  let levelToLoad = optimalLevel;
+
+  // Check here for whether optimalLevel is within max atlas size?
+  if (optimalLevel !== undefined) {
+    levelToLoad = Math.max(optimalLevel + (loadSpec.scaleLevelBias ?? 0), loadSpec.multiscaleLevel ?? 0);
+    levelToLoad = Math.max(0, Math.min(spatialDimsZYX.length - 1, levelToLoad));
+
+    if (doesSpatialDimensionFitInAtlas(spatialDimsZYX[levelToLoad], loadSpec.maxAtlasEdge)) {
+      return levelToLoad;
+    }
+  }
+
+  // Level to load could not be loaded due to atlas size constraints.
+  if (levelToLoad === undefined) {
+    // No optimal level exists so choose the smallest level to report out
+    levelToLoad = spatialDimsZYX.length - 1;
+  }
+  const smallestDims = spatialDimsZYX[levelToLoad];
+  console.error(
+    `Volume is too large; no multiscale level found that fits in preferred memory footprint. Selected level ${levelToLoad}  has dimensions `,
+    smallestDims,
+    `. Max atlas edge allowed is ${loadSpec.maxAtlasEdge}.`
+  );
+  console.log("All available levels: ", spatialDimsZYX);
+  throw new VolumeLoadError(`Volume is too large; multiscale level does not fit in preferred memory footprint.`, {
```

Comment:
If they are getting this error for 3D/YZ/ZX, we may want to suggest to users that they switch to XY mode in which their data might be viewable.  I think we need to work out how to handle that in the best way.  Maybe we need a future ticket for this.  We also need to consider how to handle this gracefully on the front end (w3cv) so maybe the 3d view stays empty but they can still switch to XY and use the viewer.